samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -54,6 +54,8 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(ContainerManager.class);
   private static final String ANY_HOST = ResourceRequestState.ANY_HOST;
+  private static final String LAST_SEEN = "LAST_SEEN";
+  private static final String FORCE_RESTART_LAST_SEEN = "FORCE_RESTART_LAST_SEEN";
```

Comment:
Isnt force-restart a misnomer here?
All its really doing is issuing a stop and relying on the AM to bring it back up?

In a corner case, AM will not be able to bring it back up on the same host, rather on a different available host?

Reply:
discussed offline, its the destinationHost param which is.

**All its really doing is issuing a stop and relying on the AM to bring it back up?**
Yup that is what is it doing

**In a corner case, AM will not be able to bring it back up on the same host, rather on a different available host?**
Yes and that is documented